lib/web_ui/lib/src/engine/semantics/heading.dart

```diff
@@ -0,0 +1,23 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../dom.dart';
+import 'semantics.dart';
+
+/// Renders semantics objects as headings with the corresponding
+/// level (h1 ... h6).
+class Heading extends PrimaryRoleManager {
+  Heading(SemanticsObject semanticsObject)
+      : super.withBasics(PrimaryRole.heading, semanticsObject);
+
+  @override
+  void update() {
+    super.update();
+
+      if (semanticsObject.headingLevel != -1) {
+        semanticsObject.setAriaRole('heading');
+        semanticsObject.element.setAttribute('aria-level', semanticsObject.headingLevel);
+      }
```

Comment:
Do we need to remove `aria-level` in the `else` branch? Or default it to something. Otherwise it will keep the old value if `headingLevel` changes to -1.

Reply:
I prefer the idea of setting a default one (with value 1), as the aria-level is required for elements with the heading role.